tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -302,14 +302,67 @@ class OptionsDictionary {
 struct Statistic;
 typedef std::function<void(ArrayRef<Statistic *> stats)> StatisticsReceiver;
 
+// Used to wrap the result of a request. There are three possibilities:
+// - The request succeeded (`value` is valid)
+// - The request was cancelled
+// - The request failed (with an `error`)
+template <typename T>
+class RequestResult {
+private:
```

Comment:
`private:` here is redundant, across the codebase we depend on the fact that immediately after a `class` opening access is private.

Reply:
removed